lib/src/matrix/decomposition/singular_value.dart

```diff
@@ -1,483 +1,951 @@
+      for (var k = 0; k < bn; k++) {
+        for (var j = 0; j < _vt.colCount; j++) {
+          var value = 0.0;
+          if (j < mn) {
+            for (var i = 0; i < _u.rowCount; i++) {
+              value += _u.getUnchecked(i, j) * input.getUnchecked(i, k);
+            }
 
-  /// Return the right singular vectors.
-  Matrix<double> get V => _v.toMatrix();
+            value /= _s[j];
+          }
 
-  /// Return the one-dimensional array of singular values.
-  List<double> get s => DataType.float.copyList(_s);
+          tmp[j] = value;
+        }
 
-  /// Return the diagonal matrix of singular values.
-  Matrix<double> get S {
-    final result =
-        Matrix(DataType.float, _n, _n, format: MatrixFormat.diagonal);
-    for (var i = 0; i < _n; i++) {
-      result.setUnchecked(i, i, _s[i]);
+        for (var j = 0; j < _vt.colCount; j++) {
+          var value = 0.0;
+          for (var i = 0; i < _vt.colCount; i++) {
+            value += _vt.getUnchecked(i, j) * tmp[i];
+          }
+
+          result.setUnchecked(j, k, value);
+        }
+      }
+
+      return result;
+    }
+    if (input is Vector<double>) {
+      // Ax=b where A is an m x n matrix
+      // Check that b is a column vector with m entries
+      if (_u.rowCount != input.count) {
+        throw ArgumentError('All vectors must have the same dimensionality.');
+      }
+
+      final result = Vector<double>(DataType.float64, _vt.colCount);
+
+      var mn = min(_u.rowCount, _vt.colCount);
+      var tmp = List.filled(_vt.colCount, 0.0);
+      double value;
+      for (var j = 0; j < _vt.colCount; j++) {
+        value = 0;
+        if (j < mn) {
+          for (var i = 0; i < _u.rowCount; i++) {
+            value += _u.getUnchecked(i, j) * input[i];
+          }
+
+          value /= _s[j];
+        }
+
+        tmp[j] = value;
+      }
+
+      for (var j = 0; j < _vt.colCount; j++) {
+        value = 0;
+        for (var i = 0; i < _vt.colCount; i++) {
+          value += _vt.getUnchecked(i, j) * tmp[i];
+        }
+
+        result[j] = value;
+      }
+
+      return result;
     }
-    return result;
   }
 
-  /// Return the two norm.
-  double get norm2 => _s[0];
+  static List<double> _columnMajorValuesOf(Matrix<num> matrix) {
+    final list = List.filled(matrix.rowCount * matrix.colCount, 0.0);
+    var index = 0;
+    for (var col = 0; col < matrix.colCount; col++) {
+      for (var row = 0; row < matrix.rowCount; row++) {
+        list[index++] = matrix.getUnchecked(row, col).toDouble();
+      }
+    }
+    return list;
+  }
 
-  /// Return the two norm condition number.
-  double get cond => _s[0] / _s[math.min(_m, _n) - 1];
+  /// Given the Cartesian coordinates (da, db) of a point p, these function return the parameters da, db, c, and s
+  /// associated with the Givens rotation that zeros the y-coordinate of the point.
+  ///
+  /// Note that Dart is a pass-by-value language, so modifiy the code little bit.
```

Comment:
Readers likely know that Dart is padding-by-value, and like me don't know what you modified.

Reply:
Removed the meaningless comment.